test/widgets/text_test.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:ui';
+
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/text.dart';
+
+void main() {
+  group('weightVariableTextStyle', () {
+    Future<void> runCheck(
+      String description,
+      {
+        required TextStyle Function(BuildContext context) styleBuilder,
+        bool platformRequestsBold = false,
+        required List<FontVariation> expectedFontVariations,
+        required FontWeight expectedFontWeight,
+      }
+    ) async {
+      testWidgets(description, (WidgetTester tester) async {
+        await tester.pumpWidget(
+          MaterialApp(
+            home: MediaQuery(
+              data: MediaQueryData(boldText: platformRequestsBold),
+              child: Builder(builder: (context) => Text('', style: styleBuilder(context))))));
+
+        final TextStyle? style = tester.widget<Text>(find.byType(Text)).style;
+        check(style)
+            .isNotNull()
+            ..has((style) => style.inherit, 'inherit').isTrue()
+            ..has((style) => style.fontVariations, 'fontVariations')
+              .isNotNull()
+              .deepEquals(expectedFontVariations)
+            ..has((style) => style.fontWeight, 'fontWeight')
+              .isNotNull()
+              .equals(expectedFontWeight);
+      });
+    }
+
+    runCheck('no context passed; default wght values',
```

Comment:
nit: the name says "run", but this function's actual behavior is to *define* a test (by calling `testWidgets`), not to run anything. (And if it did run anything, that'd be bad because it's not inside a `test` or `testWidgets` callback; whatever it does will happen unconditionally even when try to run only specific tests with e.g. `flutter test --name`.)

Let's give it a name starting with "test", following the pattern set by `testWidgets` itself. Perhaps `testWeights`?